### PR TITLE
Add support for retrieving s3 credentials from AWS credentials file

### PR DESCRIPTION
**Pull request by justfalter**
This pull request adds support for retrieving credentials from AWS credentials files (often used by [aws-cli](https://aws.amazon.com/cli/)).

As a regular user of the aws-cli, this change allows bazel-remote to leverage credentials I've already got (access key, secret key, session tokens, etc) without having to goof around with anything else. Note, this is something that minio-go already supports, I am just wiring things up.

I took the opportunity move the credentials parsing out to a `GetCredentials` function. I felt uncomfortable adding 3 more arguments to `s3proxy.New()`, which was already very long.

### CODE REVIEW

**Review by mostynb — CHANGES_REQUESTED** [review]
Looks pretty good- I just have a few minor documentation suggestions.

File: utils/flags/flags.go [hunk 1]
```diff
@@ -178,18 +186,36 @@ func GetCliFlags() []cli.Flag {
...
 			Name:    "s3.secret_access_key",
 			Value:   "",
-			Usage:   "The S3/minio secret access key to use when using S3 proxy backend.",
+			Usage:   "The S3/minio secret access key to use when using S3 proxy backend. " + s3AuthMsg(s3proxy.AuthMethodAccessKey),
 			EnvVars: []string{"BAZEL_REMOTE_S3_SECRET_ACCESS_KEY"},
 		},
+		&cli.StringFlag{
+			Name:    "s3.aws_shared_credentials_file",
+			Value:   "",
+			Usage:   "Path to the AWS credentials file. If not specified, the minio client will default to '~/.aws/credentials'. " + s3AuthMsg(s3proxy.AuthMethodAWSCredentialsFile),
```
[the remark this answers is not in the document]
Comment (justfalter) [reply utils/flags/flags.go:210]: I would specify a default value, but Minio actually takes care of locating the home directory on both windows and non-windows systems.

https://github.com/minio/minio-go/blob/master/pkg/credentials/file_aws_credentials.go#L35-L39

[the remark this answers is not in the document]
Comment (justfalter) [reply utils/flags/flags.go:210]: and https://github.com/minio/minio-go/blob/master/pkg/credentials/file_aws_credentials.go#L63-L70

**Review by mostynb — APPROVED** [review]
This looks good to me- thanks for the submission.

**Comment by mambon2** [comment]
Thank you.  I was running into this issue as well.  Would you please update container and documentation on how to pass .aws/credentials to the container?

**Comment by justfalter** [comment]
@mambon2 Docs added in #493

**Comment by mambon2** [comment]
thank you @justfalter

**Pull request merged**